api/v1beta1/types_class.go

```diff
@@ -459,6 +459,11 @@ type NetworkClassSpec struct {
 	// +optional
 	PrivateDNSZoneName string `json:"privateDNSZoneName,omitempty"`
 
+	// PrivateDNSZone enables private dns zone creation modes for a private cluster.
+	// When unspecified, it defaults to PrivateDNSZoneModeSystem which creates a private DNS zone.
```

Comment:
Can you specify what are the valid options for this field? Looks like it'll be "System or None". Can this also be validated in a webhook?